matrix_sdk/src/http_client.rs

```diff
@@ -227,46 +229,73 @@ async fn response_to_http_response(
         .expect("Can't construct a response using the given body"))
 }
 
-#[cfg(any(test, target_arch = "wasm32"))]
+#[cfg(any(target_arch = "wasm32"))]
 async fn send_request(
     client: &Client,
     request: http::Request<Vec<u8>>,
-    _: Option<Duration>,
+    _: RequestConfig,
 ) -> Result<http::Response<Vec<u8>>, HttpError> {
     let request = reqwest::Request::try_from(request)?;
     let response = client.execute(request).await?;
 
     Ok(response_to_http_response(response).await?)
 }
 
-#[cfg(all(not(test), not(target_arch = "wasm32")))]
+#[cfg(all(not(target_arch = "wasm32")))]
 async fn send_request(
     client: &Client,
     request: http::Request<Vec<u8>>,
-    timeout: Option<Duration>,
+    config: RequestConfig,
 ) -> Result<http::Response<Vec<u8>>, HttpError> {
-    let backoff = ExponentialBackoff::default();
+    let mut backoff = ExponentialBackoff::default();
     let mut request = reqwest::Request::try_from(request)?;
+    let retry_limit = Arc::new(Mutex::new(config.retry_limit));
```

Comment:
Wouldn't an atomic u64 server us better here than the lock?

Reply:
i didn't know there was an atomic u64. Definitely makes sense to use it here.

Reply:
Switched to AtomicU64.